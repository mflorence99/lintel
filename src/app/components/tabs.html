<mat-tab-group
  (selectedIndexChange)="onTabSelect($event)"
  [selectedIndex]="tabIndex">

  <mat-tab
    [label]="pluginName"
    *ngFor="let pluginName of schemas?.activePluginNames">
  </mat-tab>

</mat-tab-group>
