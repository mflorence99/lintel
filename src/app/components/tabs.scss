:host {
  display: block;
  margin-left: 1.5rem;
}
