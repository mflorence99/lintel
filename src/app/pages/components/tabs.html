<mat-tab-group
  (selectedIndexChange)="onTabSelect($event)"
  [selectedIndex]="tabIndex">

  <mat-tab
    [label]="pluginName"
    *ngFor="let pluginName of configs.pluginNames">
  </mat-tab>

</mat-tab-group>
