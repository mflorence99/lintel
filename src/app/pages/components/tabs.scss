:host {
  display: block;
  margin-left: var(--rules-gutter);
}

.unknown {
  color: var(--vscode-inputValidation-errorBorder);
}
