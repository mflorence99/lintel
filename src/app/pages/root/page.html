<mat-tree [dataSource]="dataSource" [treeControl]="treeControl">

  <mat-tree-node *matTreeNodeDef="let node" matTreeNodePadding>

    <span (click)="select(node.fileName)" class="leaf" [ngClass]="{selected: node.fileName === selection.fileName}">
      {{node.fileName}}
    </span>

  </mat-tree-node>

  <mat-tree-node *matTreeNodeDef="let node; when: hasChild" matTreeNodePadding>

    <mat-icon class="mat-icon-rtl-mirror" matTreeNodeToggle>
      {{treeControl.isExpanded(node) ? 'expand_more' : 'chevron_right'}}
    </mat-icon>

    <span (click)="select(node.fileName)" [ngClass]="{selected: node.fileName === selection.fileName}">
      {{node.fileName}}
    </span>

  </mat-tree-node>

</mat-tree>
