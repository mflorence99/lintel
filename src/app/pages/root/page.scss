:host {
  height: 100%;
}

.mat-tree-node {
  cursor: pointer;
}

.example-tree-invisible {
  display: none;
}

.example-tree ul,
.example-tree li {
  list-style-type: none;
  margin-bottom: 0;
  margin-top: 0;
}
