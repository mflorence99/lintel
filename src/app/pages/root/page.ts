import { ChangeDetectionStrategy } from '@angular/core';
import { Component } from '@angular/core';
import { ConfigsState } from '../../state/configs';
import { ElementRef } from '@angular/core';
import { FilesState } from '../../state/files';
import { Params } from '../../services/params';
import { RulesState } from '../../state/rules';
import { SchemaState } from '../../state/schema';
import { SelectionState } from '../../state/selection';

/**
 * Lintel Root
 */

@Component({
  changeDetection: ChangeDetectionStrategy.Default,
  selector: 'lintel-root',
  templateUrl: 'page.html',
  styleUrls: ['page.scss']
})

export class RootPageComponent {

  /** ctor */
  constructor(public configs: ConfigsState,
              public files: FilesState,
              private host: ElementRef,
              public params: Params,
              public rules: RulesState,
              public schema: SchemaState,
              public selection: SelectionState) {
    // TODO: must do files first
    this.files.initialize();
    this.configs.initialize();
    this.rules.initialize();
    this.schema.initialize();
  }

  /** Scroll to top */
  scrollToTop(): void {
    const scroller = this.host.nativeElement.querySelector('#scroller');
    if (scroller)
      scroller.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
  }

}
