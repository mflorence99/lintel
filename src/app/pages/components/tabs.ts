import { ChangeDetectionStrategy } from '@angular/core';
import { Component } from '@angular/core';
import { ConfigsState } from '../../state/configs';
import { EventEmitter } from '@angular/core';
import { Output } from '@angular/core';
import { Params } from '../../services/params';
import { SelectionState } from '../../state/selection';
import { SingleselectorComponent } from '../../components/singleselector';
import { ViewChild } from '@angular/core';

/**
 * Tabs component
 */

@Component({
  changeDetection: ChangeDetectionStrategy.Default,
  selector: 'lintel-tabs',
  templateUrl: 'tabs.html',
  styleUrls: ['tabs.scss']
})

export class TabsComponent {

  inMore: string[] = [];
  inTab: string[] = [];

  @ViewChild(SingleselectorComponent) 
  set more(m: SingleselectorComponent) {
    m?.registerOnChange(this.selectPluginName.bind(this));
  }

  @Output() pluginSelected = new EventEmitter<void>();

  /** ctor */
  constructor(public configs: ConfigsState,
              public params: Params,
              public selection: SelectionState) { 
    this.inTab = this.configs.pluginNames.slice(0, this.params.maxNumTabs);
    this.inMore = this.configs.pluginNames.slice(this.params.maxNumTabs);
  }

  /** Select a plugin */
  selectPluginName(pluginName: string): void {
    if (pluginName !== this.selection.pluginName) {
      this.selection.select({ pluginName });
      this.pluginSelected.emit();
    }
  }

}
