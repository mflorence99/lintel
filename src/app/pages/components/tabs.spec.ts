import { TabsComponent } from './tabs';

import { prepare } from '../page.spec';

import 'jest-extended';

import { ComponentFixture } from '@angular/core/testing';
import { TestBed } from '@angular/core/testing';

describe('TabsComponent', () => {
  let component: TabsComponent;
  let fixture: ComponentFixture<TabsComponent>;

  beforeEach(() => {
    prepare();
    fixture = TestBed.createComponent(TabsComponent);
    component = fixture.componentInstance;
    component.ngOnInit();
  });

  test('Plugins are filtered', (done) => {
    component.selection.select({
      fileName: '/home/mflorence99/lintel/package.json',
      pluginName: 'vue',
      category: 'Best Practices'
    });
    component.filter.filterRuleName('tests');
    component.utils.nextTick(() => {
      expect(component.selection.snapshot.pluginName).toBe(
        component.params.basePluginName
      );
      done();
    });
  });

  test('Plugin can be selected', () => {
    // NOTE: need to fake setting of file
    component.selection.select({
      fileName: '/home/mflorence99/lintel/package.json'
    });
    component.selectPluginName('@typescript-eslint');
    expect(component.selection.pluginName).toBe('@typescript-eslint');
  });

  test('Plugins are partitioned into tabs and overflow dropdown', () => {
    component.selection.select({
      fileName: '/home/mflorence99/lintel/package.json'
    });
    component.filter.filterRuleName('class');
    const which = component.whichPlugins();
    expect(which.inTab).toEqual([
      component.params.basePluginName,
      '@angular-eslint',
      '@typescript-eslint',
      'react',
      'vue'
    ]);
    expect(which.inMore).toEqual([]);
  });
});
