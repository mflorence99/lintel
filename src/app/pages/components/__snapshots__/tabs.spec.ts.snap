// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsComponent snapshot 1`] = `
<lintel-tabs
  __ngContext__={[Function LRootView]}
  actions$={[Function Actions]}
  cdf={[Function ViewRef]}
  configs={[Function ConfigsState]}
  destroy$={[Function DestroyService]}
  filter={[Function FilterState]}
  params={[Function Params]}
  selection={[Function SelectionState]}
  utils={[Function Utils]}
>
  <menu
    class="tabs"
  >
    <li
      class="tab selected"
    >
       eslint 
    </li>
    <li
      class="tab"
    >
       @angular-eslint 
    </li>
    <li
      class="tab"
    >
       @typescript-eslint 
    </li>
    <li
      class="tab"
    >
       compat 
    </li>
    <li
      class="tab"
    >
       import-splitnsort 
    </li>
    <li
      class="tab"
    >
       jest 
    </li>
    <li
      class="tab"
    >
       lodash-fp 
    </li>
    
  </menu><lintel-singleselector
    class="more"
    placeholder="More..."
  >
    <select
      autocomplete="off"
      tabindex="0"
    >
      <option
        value="null"
      >
        More...
      </option>
      
      <option
        value="node"
      >
         node 
      </option>
      <option
        value="react"
      >
         react 
      </option>
      <option
        value="react-hooks"
      >
         react-hooks 
      </option>
      <option
        value="react-redux"
      >
         react-redux 
      </option>
      <option
        value="sonarjs"
      >
         sonarjs 
      </option>
      <option
        value="unicorn"
      >
         unicorn 
      </option>
      <option
        value="vue"
      >
         vue 
      </option>
      <option
        value="unknown"
      >
         unknown 
      </option>
      
    </select>
  </lintel-singleselector>
</lintel-tabs>
`;
